### DISCUSSION TITLE
vm call error while testing

### DISCUSSION THREAD

**Discussion by Abhishek-2416** [Cyfrin/foundry-full-course-cu#1493] [Q&A] [answered]
Getting this errror when i am testing my ERC20

```
Failing tests:
Encountered 1 failing test in test/OurTokenTest.t.sol:OurTokenTest
[FAIL. Reason: unknown selector `0x98296c54` for VmCalls] testBobBalance() (gas: 7860)
```

**Comment by EngrPips** [comment]
Hello @Abhishek-2416 , more details would be needed to be able to help you. a screenshot of your test file and the actual test that you ran which resulted in this failure and any other details  that you think might help me to help you debug would be really appreciated.

**Reply by Abhishek-2416** [reply]
Could you please check my latest repo it has the specific test file and this error is coming for every forge command

**Reply by Abhishek-2416** [reply]
Here is my test code
```
// SPDX-License-Identifier: SEE LICENSE IN LICENSE
pragma solidity ^0.8.20;

import {Test} from "forge-std/Test.sol";
import {OurToken} from "../src/OurToken.sol";
import {DeployOurToken} from "../script/DeployOurToken.s.sol";

contract TestOurToken is Test {
    OurToken ourToken;

    //addresses
    address bob = makeAddr("bob");
    address alice = makeAddr("alice");

    uint256 constant STARTING_BALANCE = 100 ether;

    function setUp() public {
        DeployOurToken deployer = new DeployOurToken();
        ourToken = deployer.run();

        deal(bob, STARTING_BALANCE);
    }

    function testBobStartingBalance() public {
        assertEq(address(bob).balance, STARTING_BALANCE);
    }
}

```

**Reply by EngrPips** [reply]
@Abhishek-2416 please share me the link to your repo again, i can't remember what it is.

**Reply by EngrPips** [reply]
And test code looks normal at first glance.

**Reply by EngrPips** [reply]
Actually @Abhishek-2416 , i will suggest you run `foundryUp` then retry again as i ran the same test on my machine and it pass, picture below

<img width="1440" alt="Screenshot 2024-03-18 at 1 02 34 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/06bc6556-85d6-40b0-93d8-d1975fcd8dc9">

**Reply by Abhishek-2416** [reply, marked as answer]
i did ``` foundryup``` then did ```forge clean ``` after that it was working

**Reply by r0bchain** [reply]
> i did ` foundryup` then did `forge clean ` after that it was working

Hi @Abhishek-2416 
I  also fixed it, following your indications, thanks!

**Reply by EngrPips** [reply]
awesome.
